### DISCUSSION TITLE
Formatting with locals

### DISCUSSION THREAD

**Discussion by Maxime-Brassard** [iTwin/itwinjs-core#6322] [General]
We have a specific issue right now where a user tried using a , as a decimal separator as this is what is used in his local.
Specifically, he tried to input 2,75m which he expected to be 2 meters 75 cm.
The issue is that the formatter we are using (ImodelApp.quantityformatter) seems to treat "," as thousands separators and "." as decimal separators so the 2,75m he inputted behaved like it was 275m.
![Screenshot 2023-12-21 100641](https://github.com/iTwin/itwinjs-core/assets/125395992/f9cbae5e-21c0-4723-9d0b-00112f37547d)
Now, what I'd like to know is: if there is a way to affect these values depending on the local it's being used in or what would be the best course of action in this case.

**Comment by grigasp** [comment]
@ColinKerr, any ideas?

**Comment by ColinKerr** [comment]
Format definitions can include decimal and thousands separators.  The docs cover this (scroll down): https://www.itwinjs.org/bis/ec/ec-format/

If one is not explicitly specified we pull the one from the current locale.

The backend has no way to know the frontends locale when on the web so pulling from the current locale will only work in an electron app.

To support this on the web we'll have to define some way for the frontend to pass back the separators to use.

**Comment by hl662** [comment]
A [PR ](https://github.com/iTwin/itwinjs-core/pull/6387) addressing this discussion has been created. The results of the investigation are posted in issue #6362
